build/controller_primary.sh

```diff
@@ -14,73 +15,93 @@ GERRIT_REFSPEC=%%GERRIT_REFSPEC%%
+  sed -i "s/__ENVIRONMENT_VERSION__/$environment_version/g" $openstack_user_config
+  sed -i "s/__EXTERNAL_VIP_IP__/%%EXTERNAL_VIP_IP%%/g" $openstack_user_config
+  sed -i "s/__CLUSTER_PREFIX__/%%CLUSTER_PREFIX%%/g" $openstack_user_config
+
+  if [ "$DEPLOY_SWIFT" = "yes" ]; then
+    curl -o $swift_config "${raw_url}/%%HEAT_GIT_VERSION%%/swift.yml"
+    sed -i "s/__CLUSTER_PREFIX__/%%CLUSTER_PREFIX%%/g" $swift_config
 
-cd /root/os-ansible-deployment
+    sed -i "s/#\(glance_swift_store_auth_address\): .*/\1: '{{ keystone_service_internalurl }}'/" $user_variables
+    sed -i "s/#\(glance_swift_store_user\): .*/\1: 'service:glance'/" $user_variables
+    sed -i "s/#\(glance_swift_store_key\): .*/\1: '{{ glance_service_password }}'/" $user_variables
+    sed -i "s/#\(glance_swift_store_region\): .*/\1: RegionOne/" $user_variables
+  else
+    sed -i "s/#\(glance_swift_store_auth_address\): .*/\1: '{{ rackspace_cloud_auth_url }}'/" $user_variables
+    sed -i "s/#\(glance_swift_store_user\): .*/\1: '{ rackspace_cloud_tenant_id }}:{{ rackspace_cloud_username }}'/" $user_variables
+    sed -i "s/#\(glance_swift_store_key\): .*/\1: '{{ rackspace_cloud_password }}'/" $user_variables
+    sed -i "s/#\(glance_swift_store_region\): .*/\1: %%GLANCE_SWIFT_STORE_REGION%%/g" $user_variables
+
+    test -f $swift_config && rm $swift_config
+  fi
+
+  scripts/pw-token-gen.py --file $user_secrets
+popd
+
+pushd rpcd
+  cp -a etc/openstack_deploy/* $config_dir
+
+  sed -i "s/\(maas_notification_plan\): .*/\1: npTechnicalContactsEmail/" ${config_dir}/user_extras_variables.yml
+  sed -i "s/\(lb_name\): .*/\1: %%CLUSTER_PREFIX%%-node3/" ${config_dir}/user_extras_variables.yml
+  sed -i "s@\(rackspace_cloud_auth_url\): .*@\1: %%RACKSPACE_CLOUD_AUTH_URL%%@" ${config_dir}/user_extras_variables.yml
+  sed -i "s/\(rackspace_cloud_tenant_id\): .*/\1: %%RACKSPACE_CLOUD_TENANT_ID%%/" ${config_dir}/user_extras_variables.yml
+  sed -i "s/\(rackspace_cloud_username\): .*/\1: %%RACKSPACE_CLOUD_USERNAME%%/" ${config_dir}/user_extras_variables.yml
+  sed -i "s/\(rackspace_cloud_password\): .*/\1: %%RACKSPACE_CLOUD_PASSWORD%%/" ${config_dir}/user_extras_variables.yml
+  sed -i "s/\(rackspace_cloud_api_key\): .*/\1: %%RACKSPACE_CLOUD_API_KEY%%/" ${config_dir}/user_extras_variables.yml
+
+  ${checkout_dir}/rpc-openstack/os-ansible-deployment/scripts/pw-token-gen.py --file ${config_dir}/user_extras_secrets.yml
+popd
 
 # here we run ansible using the run-playbooks script in the ansible repo
 if [ "%%RUN_ANSIBLE%%" = "True" ]; then
-  scripts/bootstrap-ansible.sh
+  cd ${checkout_dir}/rpc-openstack/os-ansible-deployment
   scripts/run-playbooks.sh
-fi
-if [ "%%RUN_TEMPEST%%" = "True" ]; then
-  export TEMPEST_SCRIPT_PARAMETERS="%%TEMPEST_SCRIPT_PARAMETERS%%"
-  scripts/run-tempest.sh
+  if [ "$DEPLOY_MONITORING" = "yes" ]; then
+    cd ${checkout_dir}/rpc-openstack/rpcd/playbooks
+    openstack-ansible repo-build.yml
```

Comment:
This caught me out. I had DEPLOY_MONITORING != 'yes', so I could run bits of the monitoring setup manually but none of the repo stuff was configured so the pip install failed. What do you think to separating this out so there is a separate rpc setup flag for the repos?